stream/src/public_key/nonce.rs

```diff
@@ -3,52 +3,45 @@ use chacha20poly1305::Nonce;
 
 /// A struct that holds the nonce information.
 ///
-/// The nonce contains:
-/// - `dialer`: a boolean that is true if the nonce is for the dialer side.
-/// - `iter` and `seq`: combined, an 80-bit value that can be incremented.
+/// Holds a `u128` value that is incremented by 2 each time the nonce is incremented.
+/// The least-significant bit is used to indicate the dialer side and does not change.
+///
+/// Is able to be incremented up-to 96 bits (12 bytes) before overflowing.
 pub struct Info {
-    dialer: bool,
-    iter: u16,
-    seq: u64,
+    counter: u128,
 }
 
 impl Info {
     pub fn new(dialer: bool) -> Self {
         Self {
-            dialer,
-            iter: 0,
-            seq: 0,
+            counter: if dialer { 1 } else { 0 },
         }
     }
 
     /// Increments the nonce.
     ///
-    /// `seq` holds the least significant 64 bits of the nonce, and `iter` holds the most significant 16 bits.
-    /// An error is returned if-and-only-if the nonce overflows.
+    /// Counter is incremented by 2, maintaining the dialer bit.
+    /// An error is returned if-and-only-if the nonce overflows 96 bits.
     pub fn inc(&mut self) -> Result<(), Error> {
-        if self.seq == u64::MAX {
-            if self.iter == u16::MAX {
-                return Err(Error::NonceOverflow);
-            }
-            self.iter += 1;
-            self.seq = 0;
-            return Ok(());
+        const OVERFLOW: u128 = 1 << 96;
```

Comment:
I think we should define this outside the func? Not sure if it makes a difference to the compiler but seems like it would?

Reply:
It doesn't matter to the compiler but I can pull it out for clarity.

Sometimes in more complex files it could make sense to keep it in the scope of a certain function, especially if it's defined differently per-function